Services/Transaction/Storage/Doctrine.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/*
+ * This file is part of the ecentria group, inc. software.
+ *
+ * (c) 2016, ecentria group, inc.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Ecentria\Libraries\EcentriaRestBundle\Services\Transaction\Storage;
+
+use Ecentria\Libraries\EcentriaRestBundle\Entity\Transaction as TransactionEntity,
+    Ecentria\Libraries\EcentriaRestBundle\Model\Transaction as TransactionModel,
+    Ecentria\Libraries\EcentriaRestBundle\Converter\EntityConverter;
+
+use Doctrine\Common\Collections\ArrayCollection,
+    Doctrine\Common\Persistence\ManagerRegistry,
+    Doctrine\Common\Util\ClassUtils;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Doctrine Transaction Storage
+ *
+ * @author Artem Petrov <artem.petrov@opticsplanet.com>
+ */
+class Doctrine implements TransactionStorageInterface {
+
+    /**
+     * Registry
+     *
+     * @var ManagerRegistry
+     */
+    private $registry;
+
+    /**
+     * Request
+     *
+     * @var Request
+     */
+    private $request;
+
+    /**
+     * Entity Converter
+     *
+     * @var EntityConverter
+     */
+    private $entityConverter;
+
+    /**
+     * Constructor.
+     *
+     * @param ManagerRegistry $registry
+     * @param EntityConverter $entityConverter
+     */
+    public function __construct(ManagerRegistry $registry, EntityConverter $entityConverter) {
+        $this->registry = $registry;
+        $this->entityConverter = $entityConverter;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function write(TransactionModel $transaction) {
+        $entity = $this->buildEntity($transaction);
+        $className = class_exists('Doctrine\Common\Util\ClassUtils') ? ClassUtils::getClass($entity) : get_class($entity);
+        $entityManager = $this->registry->getManagerForClass($className);
+
+        $entityManager->clear();
+        $entityManager->persist($entity);
+        $entityManager->flush();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function read($id) {
+        $values = [
+            'class'     => 'Ecentria\Libraries\EcentriaRestBundle\Entity\Transaction',
+            'converter' => 'ecentria.api.converter.entity',
+            'name'      => 'transactionEntity'
+        ];
+
+        $configuration = new ParamConverter($values);
+        $this->entityConverter->apply($this->request, $configuration);
```

Comment:
@inferont, I had to pass Request object to the Storage in order to re-use EntityConverter::apply method.
That's not good. I'm thinking to use `$entityManager->find('Ecentria\Libraries\EcentriaRestBundle\Entity\Transaction', $id);` instead